src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -96,6 +96,10 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+    	if (!path.toFile().exists()) {
+    		throw new StorageOperationException("File not found: " + path);
```

Comment:
You have thrown the exception in the right place, but you have not handled it properly. The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception